source/Engine.cpp

```diff
@@ -2265,7 +2261,10 @@ void Engine::DoCollisions(Projectile &projectile)
 			// "safe" weapon.
 			Point hitPos = projectile.Position() + range * projectile.Velocity();
 			bool isSafe = weapon.IsSafe();
-			for(Body *body : shipCollisions.Circle(hitPos, blastRadius))
+			vector<Body *> blastCollisions;
+			blastCollisions.reserve(ships.size());
```

Comment:
How about `max(32, ships.size()`? I expect that we will typically not have more than 32 ships caught in a blast radius.

Reply:
```suggestion
			blastCollisions.reserve(min(32, ships.size()));
```
I assume you meant `min` here as well.

Reply:
Let's just go for 32. Blast radiusses are typically quite local.